snooty/rstparser.py

```diff
@@ -998,10 +998,24 @@ def make_docutils_directive_handler(
     name: str,
     options: Dict[str, object],
 ) -> Type[docutils.parsers.rst.Directive]:
+    optional_args = 0
+    required_args = 0
+
+    argument_type = directive.argument_type
+    if argument_type:
+        if (
+            isinstance(argument_type, specparser.DirectiveOption)
+            and argument_type.required
+        ):
+            required_args = 1
+        else:
+            optional_args = 1
```

Comment:
This seems to make arguments optional everywhere except where they're explicitly required… Is it really only versionchanged and versionadded where arguments should be required? Don't we need an argument for `option`, `program`, and `code-block`?

Reply:
Oh wait, this is making sure it doesn't *crash*, not that we don't complain about it. NEVER MIND, I UNDERSTAND.

Reply:
So right now all arguments are optional. This change allows us to opt things into being required. `option` and `program` are rstobjects, which this change forces to be argument required.